src/Interface/HasOrder/Instance.agda

```diff
@@ -2,29 +2,62 @@
 
 module Interface.HasOrder.Instance where
 
-open import Prelude            using (_≡_)
-open import Data.Integer as ℤ  using (ℤ)
-open import Data.Nat as ℕ      using (ℕ)
-open import Interface.HasOrder
-
-import Data.Integer.Properties as IntProp
-import Data.Nat.Properties as NatProp
-
-instance
-  preoInt : HasPreorder ℤ _≡_
-  preoInt = record { _≤_ = ℤ._≤_; isPreorder = IntProp.≤-isPreorder }
+open import Prelude
 
-  leqInt : HasPartialOrder ℤ _≡_
-  leqInt = record { hasPreorder = preoInt; antisym = IntProp.≤-antisym }
+private module Nat where
+  open import Data.Nat public
+  open import Data.Nat.Properties public
 
-  DecLeqInt : HasDecPartialOrder ℤ _≡_
-  DecLeqInt = record { hasPartialOrder = leqInt ; _≤?_ = ℤ._≤?_ }
+private module Int where
+  open import Data.Integer public
+  open import Data.Integer.Properties public
 
-  preoNat : HasPreorder ℕ _≡_
-  preoNat = record { _≤_ = ℕ._≤_; isPreorder = NatProp.≤-isPreorder }
+open import Interface.DecEq
+open import Interface.Decidable.Instance
+open import Interface.HasOrder
 
-  leqNat : HasPartialOrder ℕ _≡_
-  leqNat = record { hasPreorder = preoNat; antisym = NatProp.≤-antisym }
+instance
+  ℕ-hasPreorder : HasPreorder
+  ℕ-hasPreorder = record
+    { Nat
+    ; ≤⇔<∨≈ = λ {a b} → mk⇔
+              (λ a≤b → case (a ≟ b) of λ where (yes p) → inj₂ p ; (no ¬p) → inj₁ (Nat.≤∧≢⇒< a≤b ¬p))
+               λ where (inj₁ a<b) → Nat.<⇒≤ a<b ; (inj₂ a≡b) → Nat.≤-reflexive a≡b
+    }
+
+  ℕ-hasPartialOrder : HasPartialOrder
+  ℕ-hasPartialOrder = record { hasPreorder = ℕ-hasPreorder ; ≤-antisym = Nat.≤-antisym }
+
+  ℕ-Dec-≤ = Decidable²⇒Dec Nat._≤?_
+  ℕ-Dec-< = Decidable²⇒Dec Nat._<?_
+
+  ℕ-hasDecPartialOrder : HasDecPartialOrder {A = ℕ}
+  ℕ-hasDecPartialOrder = record {}
+
```

Comment:
@omelkonian, unfortunately, if I simply remove it, I get a "No instance of type HasDecPartialOrder was found in scope" [here](https://github.com/input-output-hk/formal-ledger-specifications/blob/5147f8d1445361bfce0bbeb8a7fe446a92c5d552/src/Ledger/Epoch.agda#L69).  Was there something else you had in mind that I should have replace lines 34 and 35 with, instead of just removing them?

Reply:
OK, that makes sense, when we bundle several constraints together in a "dummy" record such as `HasDecPartialOrder` (i.e. where all fields are instance fields -- `{{ <fieldName> }} :: <type>`), it turns out we need to declare the corresponding "dummy" instances that always have the value `record {}`.